### ISSUE TITLE
Issues with Voxelization

### ISSUE THREAD

**Issue by digital-idiot** [mit-han-lab/spvnas#55]
Here I generate a point cloud having `XY` coordinates from a regular grid and initialized with random `Z` values. I removed some points (points having `Z` value less than 26, which should be roughly ~10% of all points in the chosen grid) to simulate the sparsity. For simplicity, I used the `Z` values as the features. Consider the following code snippet:

```python
import numpy as np
import torch
from torchsparse import PointTensor
from spvnas.core.models.utils import *

# Generate Grid Coordinates
x, y = np.meshgrid(np.arange(1000), np.arange(1200), indexing='ij')
x, y = x.ravel(), y.ravel()

# Initialize with random Z values
z = np.random.randint(0, 256, x.shape)

# Remove some points for realistic scenario  
xx = x[z>25]
yy = y[z>25]
zz = z[z>25]

# Using Z values as features for testing
f = torch.from_numpy(np.stack((zz,), -1).astype(float))

# Create tensor of coordinayes
pc = torch.from_numpy(np.stack((xx, yy, zz), -1).astype(float))

# Create PointTensor
pt = PointTensor(feat=f, coords=pc)

# Initial Voxelization
st = initial_voxelize(z=pt, init_res=1, after_res=5)

# Voxelization
vt = point_to_voxel(st, pt)

```
When I check the content of `st` and `vt`, the coordinates as well as the features are zero tensors, i. e. each of the following statements returns `True`

```python
torch.all(st.C==0)  # Returns True
torch.all(st.F==0)  # Returns True
torch.all(vt.C==0)  # Returns True
torch.all(vt.F==0)  # Returns True
```
> **Am I missing something?**

**Comment by digital-idiot** [comment]
It seems the invocation of `spvoxelize` function inside `initial_voxelize` producing the zero tensor, irrespective of the input. I am not sure why. The line numbers are provided below:

https://github.com/mit-han-lab/spvnas/blob/f292399ca08fb7decc9e9b718c322000e3e807e3/core/models/utils.py#L24
https://github.com/mit-han-lab/spvnas/blob/f292399ca08fb7decc9e9b718c322000e3e807e3/core/models/utils.py#L27

**Comment by zhijian-liu** [comment]
It seems that you are using CPU for the inference. Could you try to run the inference on CUDA instead? Thanks!

**Comment by digital-idiot** [comment]
@zhijian-liu I am really sorry to miss this obvious point. The main objective of your implementation was to take advantage of the GPU hashing. When I was debugging, I observed both `sphash` and `sphashquery` both worked well by default in the CPU, so it kind of skipped my mind. It works perfectly with `CUDA`.

---

If anyone wondering, the correct version of the sample code is as follows:

```python
import numpy as np
import torch
from torchsparse import PointTensor
from spvnas.core.models.utils import *

# Make sure CUDA is available
assert torch.cuda.is_available()

# Generate Grid Coordinates
x, y = np.meshgrid(np.arange(1000), np.arange(1200), indexing='ij')
x, y = x.ravel(), y.ravel()

# Initialize with random Z values
z = np.random.randint(0, 256, x.shape)

# Remove some points for realistic scenario  
xx = x[z>25]
yy = y[z>25]
zz = z[z>25]

# Using Z values as features for testing
f = torch.from_numpy(np.stack((zz,), -1).astype(float))
f = f.cuda()

# Create tensor of coordinayes
pc = torch.from_numpy(np.stack((xx, yy, zz), -1).astype(float))
pc = pc.cuda()

# Create PointTensor
pt = PointTensor(feat=f, coords=pc)

# Initial Voxelization
st = initial_voxelize(z=pt, init_res=1, after_res=5)

# Voxelization
vt = point_to_voxel(st, pt)
```

**Comment by zhijian-liu** [comment]
Thanks for sharing the sample code! Currently, there are some bugs in the CPU inference, especially for voxelization and devoxelization. Thanks for pointing this issue out. We will take a look at this in the future (but probably of low priority as most inference should be on GPU).
